src/main/webapp/WEB-INF/web.xml

```diff
@@ -34,6 +44,16 @@
         <url-pattern>/products/*</url-pattern>
     </servlet-mapping>
 
+    <servlet-mapping>
+        <servlet-name>productPriceHistory</servlet-name>
+        <url-pattern>/products/price-history/*</url-pattern>
```

Comment:
not sure about /products
you are not displaying anything about "products", so please refactor to smth like /product/price-history/* or /price-history/*
you chose correct naming for the servlet name, but the url and its purpose now look confusing